docs/source/reference/translation.rst

```diff
@@ -0,0 +1,167 @@
+    from pytorch_lightning import Trainer
+    from flash import download_data
+    from flash.text import TranslationData, TranslationTask
+
+    # 2. Load the model from a checkpoint
+    model = TranslationTask.load_from_checkpoint("https://flash-weights.s3.amazonaws.com/translation_model_en_ro.pt")
+
+    # 3. Create dataset from file
+    datamodule = TranslationData.from_file(
+        predict_file="data/wmt_en_ro/predict.csv",
+        input="input",
+    )
+
+    # 4. generate translations
+    predictions = Trainer().predict(model, datamodule=datamodule)
+    print(predictions)
+
+For more advanced inference options, see :ref:`predictions`.
+
+-----
+
+**********
+Finetuning
+**********
+
+Say you want to finetune to your own translation data. We use the English/Romanian WMT16 dataset as an example which contains a ``train.csv`` and ``valid.csv``, structured like so:
+
+.. code-block::
+
+    input,target
+    "Written statements and oral questions (tabling): see Minutes","Declaraţii scrise şi întrebări orale (depunere): consultaţi procesul-verbal"
+    "Closure of sitting","Ridicarea şedinţei"
+    ...
+
+In the above the input/target columns represent the English and Romanian translation respectively.
+
+All we need is three lines of code to train our model!
+
+.. code-block:: python
+
+    # import our libraries
+    import flash
+    from flash import download_data
+    from flash.text import TranslationData, TranslationTask
+
+    # 1. Download data
+    download_data("https://pl-flash-data.s3.amazonaws.com/wmt_en_ro.zip", 'data/')
+
+    # Organize the data
+    datamodule = TranslationData.from_files(
+        train_file="data/wmt_en_ro/train.csv",
+        valid_file="data/wmt_en_ro/valid.csv",
+        test_file="data/wmt_en_ro/test.csv",
+        input="input",
+        target="target",
+    )
+
+    # 2. Build the task
+    model = TranslationTask()
+
+    # 4. Create trainer
+    trainer = flash.Trainer(max_epochs=5, gpus=1, precision=16)
+
+    # 5. Finetune the task
+    trainer.finetune(model, datamodule=datamodule)
+
+    # 6. Save trainer task
+    trainer.save_checkpoint("translation_model_en_ro.pt")
+
+----
+
+To run the example:
+
+.. code-block:: bash
+
+    python flash_examples/finetuning/translation.py
+
+
+------
+
+*********************
+Changing the backbone
+*********************
+By default, we use `mBART <https://github.com/pytorch/fairseq/tree/master/examples/mbart/>`_ model for translation. You can change the model run by passing in the backbone parameter.
+
+.. note:: When changing the backbone, make sure you pass in the same backbone to the Task and the Data object! Since this is a Seq2Seq task, make sure you use a Seq2Seq model.
+
+.. code-block:: python
+
+    datamodule = TranslationData.from_files(
+        train_file="data/wmt_en_ro/train.csv",
+        valid_file="data/wmt_en_ro/valid.csv",
+        test_file="data/wmt_en_ro/test.csv",
+        input="input",
+        target="target",
```

Comment:
shall this be default?

Reply:
target isn't default unfortunately, as this would break prediction support! can refactor in the future